Update prod (#4432)
OpenLiberty/blogs: posts/2025-07-01-25.0.0.7-beta.adoc

- link:https://github.com/eclipse/microprofile-health/releases/tag/4.0[MicroProfile Health 4.0]
- link:https://github.com/eclipse/microprofile-telemetry/releases/tag/2.1[MicroProfile Telemetry 2.1]
- link:https://github.com/eclipse/microprofile-open-api/releases/tag/4.1[MicroProfile OpenAPI 4.1]
- link:https://github.com/eclipse/microprofile-jwt-auth/releases/tag/2.1[MicroProfile JWT Authentication 2.1]
- link:https://github.com/eclipse/microprofile-rest-client/releases/tag/4.0[MicroProfile Rest Client 4.0]

In this release, MicroProfile Telemetry 2.1 and MicroProfile OpenAPI 4.1 have updates while the other five specifications remain unchanged.

To enable MicroProfile 7.1, add the following feature to your `server.xml`: 

```xml
<featureManager>
    <feature>microProfile-7.1</feature>
</featureManager>
```

// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31805
// Contact/Reviewer: yasmin-aumeeruddy
// // // // // // // // 
[#telemetry]
== MicroProfile Telemetry 2.1

> MicroProfile Telemetry 2.1(`mpTelemetry-2.1`) provides developers with updated Open Telemetry technology as the feature now uses OpenTelemetry v1.48.0, which was updated from v1.39.0. The previous exposed experimental APIs are now stable.

Reviewer: ```suggestion
MicroProfile Telemetry 2.1 (`mpTelemetry-2.1`) provides developers with updated Open Telemetry technology. The feature now uses OpenTelemetry v1.48.0, which was updated from v1.39.0. The previously exposed experimental APIs are now stable.
```

```xml
<featureManager>
   <feature>mpTelemetry-2.1</feature>
</featureManager>
```

To use the OpenTelemetry API, it is no longer required to add the `apiTypeVisibility="+third-party"` attribute to the `classloader` element of the server.xml. This also means that it is now possible to deploy an application that uses the OpenTelemetry API by dropping it into the dropins directory.

For more information about MicroProfile Telemetry, see the following links:

> - https://github.com/eclipse/microprofile-telemetry[MicroProfile Telemetry]
> - https://github.com/open-telemetry/opentelemetry-specification[specification]
> - https://openliberty.io/docs/latest/docs/latest/microprofile-telemetry.html[documentation]

Reviewer: ```suggestion
- https://github.com/eclipse/microprofile-telemetry[MicroProfile Telemetry]
- https://github.com/open-telemetry/opentelemetry-specification[OpenTelemetry specification]
- https://openliberty.io/docs/latest/microprofile-telemetry.html[MicroProfile Telemetry documentation for Open Liberty]
```

    
    
// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31811
// Contact/Reviewer: Azquelt
// // // // // // // // 
[#openapi]
== MicroProfile OpenAPI 4.1

> OpenAPI is a standardized way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps to generate and serve OpenAPI documentation for REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people who uses API in production.

Reviewer: ```suggestion
OpenAPI is a standardized way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps to generate and serve OpenAPI documentation for REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people who use API in production.
```

Author: For some reason, it didn't turn up in the grammar checker also.

To use the new MicroProfile OpenAPI 4.1 feature, first add the feature to your `server.xml`:

```xml
<featureManager>
    <feature>mpOpenAPI-4.1</feature>
</featureManager>
```

> Next deploy any rest application. Then, in a web browser, go to `/openapi` on the Liberty server to view the YAML OpenAPI document, or go to `/openapi/ui` for a user interface to browse through the documentation. To enhance the documentation, use link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[annotations] in code to add things like textual descriptions to each method. The annotations are available as a maven dependency:

Reviewer: ```suggestion
Next, deploy any REST application. Then, in a web browser, go to `/openapi` on the Liberty server to view the YAML OpenAPI document, or go to `/openapi/ui` for a user interface to browse through the documentation. To enhance the documentation, use link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[annotations] in code to add things like textual descriptions to each method. The annotations are available as a maven dependency:
```
